src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/NestedFieldQueryIT.java

```diff
@@ -170,6 +172,107 @@ public void singleCondition() throws IOException {
+            hits(
+                hit(
+                    textMatch("title", null),
+                    innerHits("projects.address",
+                        hit(
+                            textMatch("city", "Seattle")
+                        ),
+                        hit(
+                            textMatch("city", "Boston")
+                        ),
+                        hit(
+                            textMatch("city", "Chicago")
+                        )
+                    )
+                ),
+                hit(
+                    textMatch("title", "Software Eng 2"),
+                    innerHits("projects.address",
+                        hit(
+                            textMatch("city", "Dallas")
+                        )
+                    )
+                )
+            )
+        );
+    }
+
+    @Test
+    public void deepSelectAll() throws IOException {
+        assertThat(
+            execute("SELECT * FROM elasticsearch-sql_test_index_employee_nested e, e.projects p, p.address a"),
+            hits(
+                hit(
+                    textMatch("name", "Bob Smith"),
+                    innerHits("projects",
+                        hit(
+                            textMatch("name", "AWS Redshift Spectrum querying"),
+                            innerHits("projects.address",
+                                hit(
+                                    textMatch("city", "Seattle"),
+                                    textMatch("state", "WA")
+                                ),
+                                hit(
+                                    textMatch("city", "Boston"),
+                                    textMatch("state", "MA")
+                                )
+                            )
+                        ),
+                        hit(
+                            textMatch("name", "AWS Redshift security"),
+                            innerHits("projects.address",
+                                hit(
+                                    textMatch("city", "Chicago"),
+                                    textMatch("state", "IL")
+                                )
+                            )
+                        )
+                    )
+                ),
+                hit(
+                    textMatch("name", "Jane Smith"),
+                    innerHits("projects",
+                        hit(
+                            textMatch("name", "AWS Hello security"),
+                            innerHits("projects.address",
+                                hit(
+                                    textMatch("city", "Dallas"),
+                                    textMatch("state", "TX")
+                                )
+                            )
+                        )
+                    )
+                )
+            )
+        );
+    }
+
+    @Test
+    @Ignore("Deep query with WHERE condition is not fully supported")
```

Comment:
Please also add a test (could also be ignored) with group by

Reply:
Haven't figured out the right DSL for `GROUP BY`, but I will look into that.